### PR TITLE
*: add a guide to dApp development

### PR DESCRIPTION
**Pull request by AnnaShaleva**
An example of tiny dApp is added with proper links and comments.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -892,14 +892,14 @@ The JSON result is:
 The `stack` field contains now `2` integer value, so the counter was incremented as we expected.
 
 ## Workshop. Part 3
-In this part we'll know about NEP5 token standard and try to write, deploy and invoke more complicated smart contract.
```
Comment (roman-khimov) [inline README.md:895]: :astonished:

File: README.md [hunk 2]
```diff
@@ -149,6 +149,26 @@ Private net -- it’s the private one which you can run locally. Testnet and Mai
 Neo has a nice monitor where you can find particular node running in the blockchain network.
 [Neo Monitor](http://monitor.cityofzion.io/)
 
+## Workshop. Content
+The workshop contains a guide, examples and tips on Neo smart contracts development
+and dApps development for Neo ecosystem using development kit offered by NeoGo.
+The workshop is devided into multiple parts:
```
Comment (roman-khimov) [inline README.md]: divided

File: README.md [hunk 3]
```diff
@@ -149,6 +149,26 @@ Private net -- it’s the private one which you can run locally. Testnet and Mai
...
+1. [**Preparation.**](#workshop-preparation) Learn how to run a Neo private network,
+  transfer some funds from multi-signature account to a simple signature account
+  using NeoGo CLI and check the balance via JSON RPC call.
+2. [**Part 1.**](#workshop-part-1) Compile, inspect, deploy and invoke simple
+  `Hello, world!` smart contract written in Go.
+3. [**Part 2.**](#workshop-part-2) Investigate Neo JSON RPC protocol and NeoGo CLI
+   utilities to retrieve information from Neo RPC nodes. Get acquainted with the
+   concept of a smart contract storage. Compile, deploy and invoke contract that
+   demonstrates the how to use its storage.
```
Comment (roman-khimov) [inline README.md]: s/the//

File: README.md [hunk 4]
```diff
@@ -1529,6 +1549,63 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
 $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.json 9042814f07d65d2b835fa1f07d21c22c6e1cbdf7 delete my_second_domain -- NbrUYaZgyhSkNoRo9ugRyEMdUZxrhkNaWB
 ```
 
+## Workshop. Part 5
+
+In this part we'll take a look at the example of a decentralized application for the
+Neo ecosystem written in Go. The example demonstrates how to use NeoGo RPC Client
+and a various set of helpers to work with wallets, deploy and invoke smart
```
Comment (roman-khimov) [inline README.md]: and a set
work with wallets, deployments, invocations, data retrieval and such

File: README.md [hunk 5]
```diff
@@ -1529,6 +1549,63 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
...
+contracts, retrieve data from the blockchain, handle blockchain events and create
+auto-generated smart contract RPC bindings.
+
+#### Step #1
+
+Ensure that private network created at the [Preparation part](#workshop-preparation)
+step is up and running. We'll also need a `NbrUYaZgyhSkNoRo9ugRyEMdUZxrhkNaWB`
+account from the [wallet](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/my_wallet.json)
```
Comment (roman-khimov) [inline README.md]: Links to `master` are always dangerous.

File: README.md [hunk 6]
```diff
@@ -1529,6 +1549,63 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
...
+* [`unwrap` helper package](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient/unwrap) for test invocation results unwrapping.
+* NeoGo JSON RPC [Notification subsystem](https://github.com/nspcc-dev/neo-go/blob/master/docs/notifications.md) usage example. Subscribe for a various set of blockchain node events and receive notifications over web-socket channel.
+* NeoGo `Invoker` package usage example. Perform test invocation of smart contract, script or verification method with the powerful [Invoker API](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient/invoker).
+* How to work with witnesses and [witness scopes](https://neospcc.medium.com/thou-shalt-check-their-witnesses-485d2bf8375d).
+* [Historic invocations functionality](https://github.com/nspcc-dev/neo-go/blob/master/docs/rpc.md#invokecontractverifyhistoric-invokefunctionhistoric-and-invokescripthistoric-calls) provided by NeoGo RPC server and client as an extensions.
```
Comment (roman-khimov) [inline README.md]: "as extensions" or just "server and client extensions"

File: README.md [hunk 7]
```diff
@@ -1529,6 +1549,63 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
...
+Ensure that private network created at the [Preparation part](#workshop-preparation)
+step is up and running. We'll also need a `NbrUYaZgyhSkNoRo9ugRyEMdUZxrhkNaWB`
+account from the [wallet](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/my_wallet.json)
+with some GAS on it. Take a look at the dApp example: [dApp.go](./dApp/dApp.go).
+The example includes all basic API usages you need to be aware of to start your
+dApp development for Neo:
+* [JSON RPC client](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient) creation, initialization and example of simple Neo JSON RPC APIs usage.
+* Neo wallet and account managing with [NeoGo `wallet` package](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/wallet).
+* [Extensions](https://github.com/nspcc-dev/neo-go/blob/master/docs/rpc.md#extensions) offered by NeoGo JSON RPC server and supported by the NeoGo RPC client.
+* NeoGo JSON RPC server [web-socket extension](https://github.com/nspcc-dev/neo-go/blob/master/docs/rpc.md#websocket-server). [Web-socket JSON RPC Client](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient#WSClient) creation, initialisation and usage example.
```
Comment (roman-khimov) [inline README.md]: initialization

File: README.md [hunk 8]
```diff
@@ -1529,6 +1549,63 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
...
+* Neo wallet and account managing with [NeoGo `wallet` package](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/wallet).
+* [Extensions](https://github.com/nspcc-dev/neo-go/blob/master/docs/rpc.md#extensions) offered by NeoGo JSON RPC server and supported by the NeoGo RPC client.
+* NeoGo JSON RPC server [web-socket extension](https://github.com/nspcc-dev/neo-go/blob/master/docs/rpc.md#websocket-server). [Web-socket JSON RPC Client](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient#WSClient) creation, initialisation and usage example. 
+* [`unwrap` helper package](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient/unwrap) for test invocation results unwrapping.
+* NeoGo JSON RPC [Notification subsystem](https://github.com/nspcc-dev/neo-go/blob/master/docs/notifications.md) usage example. Subscribe for a various set of blockchain node events and receive notifications over web-socket channel.
+* NeoGo `Invoker` package usage example. Perform test invocation of smart contract, script or verification method with the powerful [Invoker API](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient/invoker).
```
Comment (roman-khimov) [inline README.md]: `invoker` as a package, `Invoker` as a struct/interface.

File: README.md [hunk 9]
```diff
@@ -1529,6 +1549,63 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
...
+* How to work with witnesses and [witness scopes](https://neospcc.medium.com/thou-shalt-check-their-witnesses-485d2bf8375d).
+* [Historic invocations functionality](https://github.com/nspcc-dev/neo-go/blob/master/docs/rpc.md#invokecontractverifyhistoric-invokefunctionhistoric-and-invokescripthistoric-calls) provided by NeoGo RPC server and client as an extensions.
+* NeoGo `Actor` package usage example. Build, tune, sign, send and await transactions with our flexible [Actor API](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient/actor).
```
Comment (roman-khimov) [inline README.md]: actor/Actor as well.

File: README.ru.md [hunk 10]
```diff
@@ -145,6 +145,28 @@ Network-слой - один из самых важных частей ноды.
...
+частей:
+
+1. [**Подготовка.**](#workshop-preparation) Узнайте, как запустить локальную сеть Neo, перевести
+  средства на аккаунт с помощью NeoGo CLI и проверить баланс с помощью вызова
+  JSON RPC.
+2. [**Часть 1.**](#workshop-part-1) Скомпилируйте, исследуйте, задеплойте и вызовите простой
+  пример контракта `Hello, world!`, написанного на Go.
+3. [**Часть 2.**](#workshop-part-2) Познакомьтесь с протоколом Neo JSON RPC и утилитами NeoGo
+  CLI для получения информации от RPC-нод Neo. Изучите концепцию хранилища смарт
+  контрактов. Скомпилируте, задеплойте и вызовите смарт контракт, который
```
Comment (roman-khimov) [inline README.ru.md]: скомпилируйте

File: README.ru.md [hunk 11]
```diff
@@ -145,6 +145,28 @@ Network-слой - один из самых важных частей ноды.
...
+  демонстрирует вариант использования своего хранилища.
+4. [**Часть 3.**](#workshop-part-3) Узнайте больше о  стандарте токенов NEP-17 и изучите
+  пример контракта, поддерживающего стандарт NEP-17.
+5. [**Часть 4.**](#workshop-part-4) Соберите в кучку знания, полученные о смарт контрактах.
+  Скомпилируйте, задеплойте и вызовите более сложный смарт контракт.
+6. [**Часть 5.**](#workshop-part-5) Узнайте, как разработать простое децентрализованное
```
Comment (roman-khimov) [inline README.ru.md]: Вообще, не очень лайкаю императивы в таких списках, но, наверное, пойдёт.

File: README.ru.md [hunk 12]
```diff
@@ -145,6 +145,28 @@ Network-слой - один из самых важных частей ноды.
...
+## Воркшоп. Содержание
+Воркшоп содержит обучающее руководство, примеры и подсказки, помогающие начать
+разработку смарт контрактов и децентрализованных приложений для экосистемы Neo
+с помощью инструментов, предлагаемых проектом NeoGo. Воркшоп содержит несколько
+частей:
+
+1. [**Подготовка.**](#workshop-preparation) Узнайте, как запустить локальную сеть Neo, перевести
+  средства на аккаунт с помощью NeoGo CLI и проверить баланс с помощью вызова
+  JSON RPC.
+2. [**Часть 1.**](#workshop-part-1) Скомпилируйте, исследуйте, задеплойте и вызовите простой
```
Comment (roman-khimov) [inline README.ru.md]: деплой это развёртывание, но может и не очень понятно так.

File: README.ru.md [hunk 13]
```diff
@@ -1524,6 +1546,68 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
...
+#### Шаг #1
+
+Убедитесь, что локальная сеть из четырех узлов, описанная в [подготовительной части](#workshop-preparation)
+настроена и запущена. Нам также понадобится аккаунт `NbrUYaZgyhSkNoRo9ugRyEMdUZxrhkNaWB`
+из [кошелька](https://github.com/nspcc-dev/neo-go-sc-wrkshp/blob/master/my_wallet.json) с некоторым количеством токена GAS на нём. Откройте и внимательно
+просмотрите пример децентрализованного приложения: [dApp.go](./dApp/dApp.go). Пример включает
+в себя различные варианты использования базового API, которые необходимы для
+начала разработки своего собственного децентрализванного приложения в экосистеме
+Neo:
+* Создание [JSON RPC клиента](https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient), его инициализация и примеры вызовов базовых
```
Comment (roman-khimov) [inline README.ru.md]: "JSON-RPC-клиента", видимо. Или "клиента JSON-RPC"

File: README.ru.md [hunk 14]
```diff
@@ -1524,6 +1546,68 @@ $ ./bin/neo-go contract invokefunction -r http://localhost:20331 -w my_wallet.js
...
+репозиторий воркшопа и скомпилируйте демонстрационный пример контракта Storage
+из репозитория NeoGo:
+```
+$ git clone https://github.com/nspcc-dev/neo-go
+$ cd neo-go
+$ make build
+$ ./bin/neo-go contract compile -i ./examples/storage/storage.go -c ./examples/storage/storage.yml -o examples/storage/storage.nef -m ./examples/storage/storage.manifest.json
+```  
+
+#### Step #2
```
Comment (roman-khimov) [inline README.ru.md]: Степ номер два!

File: dApp/go.mod [hunk 15]
```diff
@@ -0,0 +1,23 @@
+module dApp
+
+go 1.19
```
Comment (roman-khimov) [inline dApp/go.mod]: 1.18?

File: README.md [hunk 16]
```diff
@@ -1611,6 +1611,8 @@ Thank you!
 ### Useful links
 
 * [Our basic tutorial on Medium](https://medium.com/@neospcc/%D1%81%D0%BC%D0%B0%D1%80%D1%82-%D0%BA%D0%BE%D0%BD%D1%82%D1%80%D0%B0%D0%BA%D1%82-%D0%B4%D0%BB%D1%8F-neo-769139352b65)
+* [Go smart contracts development workshop on YouTube](https://www.youtube.com/watch?v=o38fXiLG7EM)
```
Comment (roman-khimov) [inline README.md]: We can have two links here.

File: README.md (hunk 2 above)
Comment (AnnaShaleva) [reply README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (AnnaShaleva) [reply README.md]: Fixed.

File: README.md (hunk 4 above)
Comment (AnnaShaleva) [reply README.md]: Fixed.

File: README.md (hunk 5 above)
Comment (AnnaShaleva) [reply README.md]: Fixed.

File: README.md (hunk 6 above)
Comment (AnnaShaleva) [reply README.md]: Fixed.

File: README.md (hunk 7 above)
Comment (AnnaShaleva) [reply README.md]: Fixed.

File: README.md (hunk 8 above)
Comment (AnnaShaleva) [reply README.md]: Sure, fixed.

File: README.md (hunk 9 above)
Comment (AnnaShaleva) [reply README.md]: Fixed.

File: README.ru.md (hunk 11 above)
Comment (AnnaShaleva) [reply README.ru.md]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**